pkg/webhook/workloadspread/validating/workloadspread_validation.go

```diff
@@ -0,0 +1,297 @@
+		if subset.RequiredNodeSelectorTerm == nil && subset.PreferredNodeSelectorTerms == nil && subset.Tolerations == nil {
+			allErrs = append(allErrs, field.Invalid(fldPath.Index(i).Child("requiredNodeSelectorTerm"), subset.RequiredNodeSelectorTerm, "The requiredNodeSelectorTerm, preferredNodeSelectorTerms and tolerations are empty that is not valid for WorkloadSpread"))
+		} else {
+			if subset.RequiredNodeSelectorTerm != nil {
+				coreNodeSelectorTerm := &core.NodeSelectorTerm{}
+				if err := corev1.Convert_v1_NodeSelectorTerm_To_core_NodeSelectorTerm(subset.RequiredNodeSelectorTerm.DeepCopy(), coreNodeSelectorTerm, nil); err != nil {
+					allErrs = append(allErrs, field.Invalid(fldPath.Index(i).Child("requiredNodeSelectorTerm"), subset.RequiredNodeSelectorTerm, fmt.Sprintf("Convert_v1_NodeSelectorTerm_To_core_NodeSelectorTerm failed: %v", err)))
+				} else {
+					allErrs = append(allErrs, corevalidation.ValidateNodeSelectorTerm(*coreNodeSelectorTerm, fldPath.Index(i).Child("requiredNodeSelectorTerm"))...)
+				}
+			}
+
+			if subset.PreferredNodeSelectorTerms != nil {
+				corePreferredSchedulingTerms := make([]core.PreferredSchedulingTerm, 0, len(subset.PreferredNodeSelectorTerms))
+				for i, term := range subset.PreferredNodeSelectorTerms {
+					corePreferredSchedulingTerm := &core.PreferredSchedulingTerm{}
+					if err := corev1.Convert_v1_PreferredSchedulingTerm_To_core_PreferredSchedulingTerm(term.DeepCopy(), corePreferredSchedulingTerm, nil); err != nil {
+						allErrs = append(allErrs, field.Invalid(fldPath.Index(i).Child("preferredSchedulingTerms"), subset.PreferredNodeSelectorTerms, fmt.Sprintf("Convert_v1_PreferredSchedulingTerm_To_core_PreferredSchedulingTerm failed: %v", err)))
+					} else {
+						corePreferredSchedulingTerms = append(corePreferredSchedulingTerms, *corePreferredSchedulingTerm)
+					}
+				}
+
+				allErrs = append(allErrs, corevalidation.ValidatePreferredSchedulingTerms(corePreferredSchedulingTerms, fldPath.Index(i).Child("preferredSchedulingTerms"))...)
+			}
+		}
+
+		if subset.Tolerations != nil {
+			var coreTolerations []core.Toleration
+			for i, toleration := range subset.Tolerations {
+				coreToleration := &core.Toleration{}
+				if err := corev1.Convert_v1_Toleration_To_core_Toleration(&toleration, coreToleration, nil); err != nil {
+					allErrs = append(allErrs, field.Invalid(fldPath.Index(i).Child("tolerations"), subset.Tolerations, fmt.Sprintf("Convert_v1_Toleration_To_core_Toleration failed: %v", err)))
+				} else {
+					coreTolerations = append(coreTolerations, *coreToleration)
+				}
+			}
+			allErrs = append(allErrs, corevalidation.ValidateTolerations(coreTolerations, fldPath.Index(i).Child("tolerations"))...)
+		}
+
+		//TODO validate patch
+
+		//1. All subset maxReplicas must be the same type: int or percent.
+		//2. Only the last subset maxReplicas can be not specified, indicates subset replicas is not limited.
```

Comment:
I'm not sure if we should limit only the last one can be defined with no maxReplicas.
What if users need to use it with `Adaptive` strategy? Which means always use the former subset if it has enough resource, otherwise try the later subsets.

Reply:
Look at the fourth point below, When adaptive, the last subset must be not specified.

Reply:
I mean the non-last subset can also be defined with no `maxReplicas`.
For example,

```yaml
  - name: subset-a
    # ...
  - name: subset-b
    # ...
```

With Adaptive strategy, it means always use `subset-a` if it has enough resource. Otherwise, try the `subset-b`. So the former one do not need `maxReplicas`.

Reply:
subset-a maxReplicas=100% is ok?

Reply:
what if there are a,b,c three subsets?